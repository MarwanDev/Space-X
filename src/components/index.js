export { default as Navbar } from './Navbar/Navbar';
export { default as Rockets } from './Rockets/Rockets';
export { default as Missions } from './Missions/Missions';
export { default as MyProfile } from './MyProfile/MyProfile';

