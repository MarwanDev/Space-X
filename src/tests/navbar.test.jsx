import { Navbar } from "../components";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import { describe, it, beforeEach, expect } from "vitest";
import renderer from "react-test-renderer";
// import store from "../redux/store";
// import { Provider } from "react-redux";

describe("Navbar", () => {
  beforeEach(() => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
  });
  it("should render the logo and title", () => {
    const logo = screen.getByAltText("logo");
    const title = screen.getByText("Space Travelers' Hub");
    expect(logo).toBeInTheDocument();
    expect(title).toBeInTheDocument();
  });
  it("should render the correct nav links", () => {
    const rocketsLink = screen.getByText("Rockets");
    const missionsLink = screen.getByText("Missions");
    const profileLink = screen.getByText("My Profile");
    expect(rocketsLink).toHaveAttribute("href", "/");
    expect(missionsLink).toHaveAttribute("href", "/missions");
    expect(profileLink).toHaveAttribute("href", "/profile");
  });
  it("should apply the active class to the active link", () => {
    const rocketsLink = screen.getByText("Rockets");
    const missionsLink = screen.getByText("Missions");
    const profileLink = screen.getByText("My Profile");
    expect(rocketsLink).toHaveClass("active");
    expect(missionsLink).not.toHaveClass("active");
    expect(profileLink).not.toHaveClass("active");
  });
  it("renders navbar correctly", () => {
    const navbar = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      )
      .toJSON();
    expect(navbar).toMatchSnapshot();
  });
});
