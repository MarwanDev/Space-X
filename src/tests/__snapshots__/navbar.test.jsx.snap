// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > renders navbar correctly 1`] = `
<header>
  <a
    className="logo-container"
    href="/"
  >
    <img
      alt="logo"
      src="src/assets\\\\planet.png"
    />
    <h1>
      Space Travelers' Hub
    </h1>
  </a>
  <ul
    className="links-container"
  >
    <li>
      <a
        activeClassName="active"
        activeStyle={
          {
            "color": "red",
          }
        }
        aria-current="page"
        className="nav-routes active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        activeClassName="active"
        className="nav-routes"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        activeClassName="active"
        className="nav-routes"
        href="/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</header>
`;
